internal/handler/websocket.go

```diff
@@ -52,12 +52,12 @@ func WebSocket(c *gin.Context) {
 
 	// Streaming messages to the client
 	for {
-		metrics := metric.GetAllSystemMetrics()
-		//if metricsErr != nil {
-		//	log.Printf("[FAIL] | Failed to get system metrics: %v", metricsErr)
-		//	c.JSON(http.StatusInternalServerError, gin.H{"error": "Failed to get system metrics"})
-		//	return
-		//}
+		metrics, metricsErr := metric.GetAllSystemMetrics()
+		if metricsErr != nil {
+			log.Printf("[FAIL] | Failed to get system metrics: %v", metricsErr)
+			c.JSON(http.StatusInternalServerError, gin.H{"error": "Failed to get system metrics"})
+			return
+		}
```

Comment:
Oops. I set up a WebSocket connection and added error messages for handling errors, but I mistakenly implemented them as plain HTTP requests instead of WebSocket. Since we hadn’t encountered error cases before, we only just caught this problem.

Reply:
> Oops. I set up a WebSocket connection and added error messages for handling errors, but I mistakenly implemented them as plain HTTP requests instead of WebSocket. Since we hadn’t encountered error cases before, we only just caught this problem.

Shall we merge this first and then fix that in a new PR?

Reply:
> > Oops. I set up a WebSocket connection and added error messages for handling errors, but I mistakenly implemented them as plain HTTP requests instead of WebSocket. Since we hadn’t encountered error cases before, we only just caught this problem.
> 
> Shall we merge this first and then fix that in a new PR?

Yes we can merge this